### PR TITLE
Update trinity to 2.15.1

### PR DESCRIPTION
**Pull request by joshuazhuang7**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by joshuazhuang7** [comment]
@BiocondaBot please fetch artifacts

**Comment by joshuazhuang7** [comment]
@BiocondaBot please add label

File: recipes/trinity/build.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/bin/bash
 set -x -e
 
+patch -p1 < ${RECIPE_DIR}/makefile.patch
```
Comment (bgruening) [inline recipes/trinity/build.sh:4]: Why do you need to run this manually? Specifying the patch as part of meta.yaml will apply it and is more obvious for the reader of the yml file imho.

Comment (joshuazhuang7) [reply recipes/trinity/build.sh:4]: Good catch and I agree that specifying the patch in meta.yml will be easier to understand. I will push a follow-up commit with updates shortly.

**Review by bgruening — APPROVED** [review]
Thanks a bunch @joshuazhuang7

**Comment by bgruening** [comment]
@joshuazhuang7 a few of our local tests are failing with `Can't locate DB_File.pm in @INC (you may need to install the DB_File module)`. Only a few, not all. Do you know about that?

**Comment by joshuazhuang7** [comment]
I (fortunately) did not run into errors related to perl modules. A quick search of the DB_File perl module brings me to the [perl-db_file](https://anaconda.org/conda-forge/perl-db_file) conda-forge package, and we could try adding 'perl-db_file' under requirements/build.

**Comment by bgruening** [comment]
See here: https://github.com/bioconda/bioconda-recipes/pull/40337

**Pull request merged**